janus-idp/backstage-showcase
# feat(app): make learning paths configurable

## Description

Data powering the learning path link is in code itself (https://raw.githubusercontent.com/janus-idp/backstage-showcase/main/packages/app/public/learning-paths/data.json) which ideally should be configurable like home page and tech radar. https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html/getting_started_with_red_hat_developer_hub/proc-customize-rhdh-homepage_rhdh-getting-started 

## Which issue(s) does this PR fix

- Fixes 
https://issues.redhat.com/browse/RHIDP-1638 

## How to test changes / Special notes to the reviewer

As part of this PR made the data for the learning  path page configurable , that can be passed in `app-config.yaml `as a proxy. API URL will return the data in JSON in the expected format/signature and fall back to the local data if API call fails or is not configured. This is an extension to the existing configuration we have home and tech radar

https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html/getting_started_with_red_hat_developer_hub/proc-customize-rhdh-homepage_rhdh-getting-started

- Add below to app-config.yaml

```
proxy:
  endpoints:
    '/developer-hub':
      target: https://raw.githubusercontent.com/
      pathRewrite:
        '^/api/proxy/developer-hub/learning-paths': '/janus-idp/backstage-showcase/main/packages/app/public/learning-paths/data.json'
        '^/api/proxy/developer-hub/tech-radar': '/janus-idp/backstage-showcase/main/packages/app/public/tech-radar/data-default.json'
        '^/api/proxy/developer-hub': '/janus-idp/backstage-showcase/main/packages/app/public/homepage/data.json'
      changeOrigin: true
      secure: true
```

## PR acceptance criteria

Please make sure that the following steps are complete:
